java/se/sics/mspsim/core/DisAsm.java

```diff
@@ -363,34 +349,17 @@ public DbgInstruction disassemble(int pc, int[] memory, int[] reg,
       jmpOffset = (jmpOffset & 0x200) == 0 ?
         2 * jmpOffset : -(2 * (0x200 - (jmpOffset & 0x1ff)));
       String opstr = "";
-      switch(instruction & 0xfc00) {
-      case JNE:
-        opstr = "JNE";
-        break;
-      case JEQ:
-        opstr = "JEQ";
-        break;
-      case JNC:
-        opstr = "JNC";
-        break;
-      case JC:
-        opstr = "JC";
-        break;
-      case JN:
-        opstr = "JN";
-        break;
-      case JGE:
-        opstr = "JGE";
-        break;
-      case JL:
-        opstr = "JL";
-        break;
-      case JMP:
-        opstr = "JMP";
-        break;
-      default:
-        System.out.println("Not implemented instruction: " +
-                           Utils.binary16(instruction));
+      switch (instruction & 0xfc00) {
+        case JNE -> opstr = "JNE";
+        case JEQ -> opstr = "JEQ";
+        case JNC -> opstr = "JNC";
+        case JC -> opstr = "JC";
+        case JN -> opstr = "JN";
+        case JGE -> opstr = "JGE";
+        case JL -> opstr = "JL";
+        case JMP -> opstr = "JMP";
+        default -> System.out.println("Not implemented instruction: " +
+                Utils.binary16(instruction));
```

Comment:
I think this can be simplified as
```java
      String opstr = switch (instruction & 0xfc00) {
          case JNE -> "JNE";
          case JEQ -> "JEQ";
          case JNC -> "JNC";
          case JC -> "JC";
          case JN -> "JN";
          case JGE -> "JGE";
          case JL -> "JL";
          case JMP -> "JMP";
          default -> {
              System.out.println("Not implemented instruction: " + Utils.binary16(instruction));
              yield "";
          }
      };
```

Reply:
Fixed